README.markdown

```diff
@@ -125,170 +70,157 @@ You can also use this parameter to set a specific registry value as the default
 
-This type may be used outside of the `registry::value` defined type for specific, customized requirements. 
+###Public Custom Types
+* `registry_key`: Manages individual Registry keys.
+* `registry_value`: Manages individual Registry values.
 
-####Parameters in `registry_value`
+###Parameters
 
-#####`path`
+####`registry::value`:
 
-Specifies the path of the registry value to manage. For example: 'HKLM\Software\Value1' or 'HKEY_LOCAL_MACHINE\Software\Vendor\Value2'. 
+#####`key`
 
-If Puppet is running on a 64-bit system, the 32-bit Registry value can be explicitly managed using a prefix. For example: '32:HKLM\Software\Value3'.
+*Required.* Specifies a Registry key for Puppet to manage. If any of the parent keys in the path don't exist, Puppet creates them automatically. Valid options: a string containing a Registry path.
 
-#####`ensure`
+#####`data`
 
-Determines whether or not the value must exist. If not included, the module will do nothing but make sure the other parameters are met if the value is found on a node; it will not force its existence or absence. Valid values are 'present' and 'absent'.
+*Required.* Provides the contents of the specified value. Valid options: a string by default; an array if specified through the `type` parameter.
 
 #####`type`
 
-Determines the type of the Registry value data. Valid values are 'string', 'array', 'dword', 'qword', 'binary', or 'expand'.
+*Optional.* Sets the data type of the specified value. Valid options: 'string', 'array', 'dword', 'qword', 'binary', and 'expand'. Default value: 'string'.
 
-#####`data`
+#####`value`
 
-Lists the data inside the Registry value. Data should be specified as a string value, unless you have set the `type` parameter to 'array'.
+*Optional.* Determines what Registry value(s) to manage within the specified key. To set a Registry value as the default value for its parent key, name the value '(default)'. Valid options: a string. Default value: the title of your declared resource.
 
-###Defined Type: registry::service
+####`registry_key`
 
-The `registry::service` defined resource type utilizes specific Registry keys and values to manage service entries in the Microsoft service control framework. Specifically, `registry::service` manages the values in the key HKLM\System\CurrentControlSet\Services\$name\.
+#####`ensure`
 
-    registry::service { puppet:
-      ensure       => present,
-      display_name => "Puppet Agent",
-      description  => "Periodically fetches and applies configurations from a Puppet master server.",
-      command      => 'C:\PuppetLabs\Puppet\service\daemon.bat',
-    }
+*Optional.* Tells Puppet whether the key should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: if unspecified, Puppet won't enforce any other parameters unless the key is already present on a node.
```

Comment:
Users shouldn't rely on ensure having a default value. It may be optional for the registry module, but that's not the case in general, and I wouldn't recommend giving users the impression that it is optional. For example, the `file` type doesn't mention its ensure property being optional or not, it just says what the possible values are: https://docs.puppetlabs.com/references/latest/type.html#file-attribute-ensure